src/index.js

```diff
@@ -49,17 +49,28 @@ class DBLAPI extends EventEmitter {
        * @param {error} error The error
        */
 
+      this._interval = null
+
       this.client = client;
-      this.client.on('ready', () => {
+      const setup = () => {
         this.postStats()
           .then(() => this.emit('posted'))
           .catch(e => this.emit('error', e));
-        setInterval(() => {
+
+        if (this._interval) clearInterval(this._interval)
```

Comment:
If an interval exists the setup function can just return, no need to restart it.

Reply:
Cool you made it return but you're still causing a new post, move the check to the top of the function